integration/base/src/main/java/com/inrupt/client/integration/base/CoreModulesResource.java

```diff
@@ -117,50 +113,42 @@ static void setup() {
             .orElse(webIdService.getMockServerUrl() + Utils.FOLDER_SEPARATOR + MOCK_USERNAME);
 
         State.WEBID = URI.create(webidUrl);
+        State.PRIVATE_RESOURCE_PATH = PRIVATE_RESOURCE_PATH; //needed in the Mocks
         //find storage from WebID using only core module
         final Request requestRdf = Request.newBuilder(URI.create(webidUrl)).GET().build();
         final var responseRdf = client.send(requestRdf, JenaBodyHandlers.ofModel());
         final var issuers = responseRdf.body()
                 .listObjectsOfProperty(createProperty(Solid.oidcIssuer.toString()))
                 .toList();
-        issuer = issuers.get(0).toString();
+        final String issuer = issuers.get(0).toString();
         final var storages = responseRdf.body()
                 .listObjectsOfProperty(createProperty(PIM.storage.toString()))
                 .toList();
         if (!storages.isEmpty()) {
             podUrl = storages.get(0).toString();
         }
-        if (PUBLIC_RESOURCE_PATH.isEmpty()) {
-            testContainerURI = URIBuilder.newBuilder(URI.create(podUrl))
-                .path("test-" + UUID.randomUUID())
-                .path(testContainer).build();
-        } else {
-            testContainerURI = URIBuilder.newBuilder(URI.create(podUrl))
-                .path(PUBLIC_RESOURCE_PATH)
-                .path("test-" + UUID.randomUUID())
-                .path(testContainer)
+        final Session session = OpenIdSession.ofClientCredentials(
+                URI.create(issuer), //Client credentials
+                CLIENT_ID,
+                CLIENT_SECRET,
+                AUTH_METHOD);
+        localAuthClient = SolidSyncClient.getClient().session(session);
+
+        publicContainerURI = URIBuilder.newBuilder(URI.create(podUrl))
+                .path("public-core-test-" + UUID.randomUUID() + "/")
                 .build();
 
-            publicContainerURI = URIBuilder.newBuilder(URI.create(podUrl))
-                    .path(PUBLIC_RESOURCE_PATH + "/").build();
-            createAuthenticatedClient();
-            createContainer(publicContainerURI);
-            prepareACR(publicContainerURI);
-        }
+        //if a tests fails it can be that the cleanup was not properly done, so we do it here too
+        Utils.deleteContentsRecursively(localAuthClient, publicContainerURI);
```

Comment:
This line becomes redundant since you are now defining a unique new  `publicContainerURI`, so it doesn't yet exist and cannot have any contents. It is created on the next line. There may be other cases like this to check for.